org.lflang/src/org/lflang/federated/FedASTUtils.java

```diff
@@ -230,21 +229,32 @@ private static void addNetworkReceiverReaction(
         // Add the action as a trigger to the receiver reaction
         networkReceiverReaction.getTriggers().add(triggerRef);
         
+        String body = generator.generateNetworkReceiverBody(
+                networkAction,
+                sourceRef,
+                destRef,
+                receivingPortID,
+                sourceFederate,
+                destinationFederate,
+                rightBankIndex,
+                rightChannelIndex,
+                ASTUtils.getInferredType(networkAction),
+                connection.isPhysical(),
+                serializer
+        );
+        
         // Generate code for the network receiver reaction
         networkReceiverReaction.setCode(factory.createCode());
-        networkReceiverReaction.getCode().setBody(generator.generateNetworkReceiverBody(
-            networkAction,
-            sourceRef,
-            destRef,
-            receivingPortID,
-            sourceFederate,
-            destinationFederate,
-            rightBankIndex,
-            rightChannelIndex,
-            ASTUtils.getInferredType(networkAction),
-            connection.isPhysical(),
-            serializer
-        ));
+        networkReceiverReaction.getCode().setBody(body);
+        
+        // Set the STP handler to be the same as the body.
+        // FIXME: This is a hack that assumes STP violations are passed
+        // on to the reactions that are triggered by this network receiver.
+        if (coordination == CoordinationType.DECENTRALIZED) {
+            networkReceiverReaction.setStp(factory.createSTP());
+            networkReceiverReaction.getStp().setCode(factory.createCode());
+            networkReceiverReaction.getStp().getCode().setBody(body);
```

Comment:
I don't think this would do what we want unless the STP handler body manually passes down the STP violation somehow.

Reply:
I guess the issue here is that we don't want to silently invoke reactions at the wrong logical time when there is no STP handler, but then, when there is an STP violation, the network receiver, which doesn't have an STP handler, will trigger an error message even if there is a handler downstream.  I don't have a good solution for this. Suggestions?  I don't think going back to silently invoking reactions at the wrong time is the right solution.